### ISSUE TITLE
Feature: Shortest Path from Owned Principals

### ISSUE THREAD

**Issue by 0xGreen** [SpecterOps/BloodHound#680]
## Feature Description

![](https://i.ibb.co/Vpvv0vf/zkruqjp.png)

"Shortest Path from Owned Principals" cypher queries are not yet implemented in BloodhoundCE. It would be really awesome to have these. In the past at varios occasion i needed "Shortest Path from Owned Principals" queries, but one way or another the information was somehow available but  recently, a HackTheBox Machine named "Jab" the path was no where to found with Bloodhound CE, However, it was just a click away with BloodHound 4.3.1. Would like to request if these cyber queries can be added.

Also, tried some basic custom queries around "Owned Principals" with the help of chat GPT but none worked.

### Are you intending to implement this feature?

No, unfortunately, don't have that skillset

## Current Behavior

Queries are missing and there is no ways to find the attack path, the very obvious path got missed.

## Desired Behavior

Show the attack path as in the image above, which was displayed with BloodHound 4.3.1.

It would also be nice to add other cypher queries related to "Owned Principals", e.g. "Shortest Path to Domain Admin from Owned Principals" etc.

## Use Case

In my opinion, various use cases, whenever you have an owned principal, it would be nice to know where you can go from there.

**Comment by TactiFail** [comment]
Yeah this is desperately missed in the new version. Between this and the missing "Owned" icon on the graphs, it feels like a huge step backwards.

**Comment by StephenHinck** [comment]
Hey folks, we've got an upcoming surge of work that will cover several features including those mentioned related to Owned principals. Stay tuned!

**Comment by olafhartong** [comment]
just spotted this lovely PR https://github.com/SpecterOps/BloodHound/pull/795

**Comment by StephenHinck** [comment]
This will get fixed with #930 , releasing next week.
